src/Psalm/Internal/Type/Comparator/AtomicTypeComparator.php

```diff
@@ -395,10 +395,6 @@ public static function isContainedBy(
             }
 
             foreach ($input_type_part->as->getAtomicTypes() as $input_as_type_part) {
-                if ($input_as_type_part instanceof TNull && $container_type_part instanceof TNull) {
```

Comment:
TNull was already returned here: https://github.com/vimeo/psalm/pull/7229/files#diff-e17ecd10daef9f0bed66b49fc187403cae341f5a3994fdf1f776f044103a3e66L128

Reply:
That's for the container type part, right?

Reply:
yeah, so it goes `$input_as_type_part instanceof TNull && false` that is simplified to false, so the conditional is never met